impl/src/main/java/com/sun/faces/cdi/clientwindow/ClientWindowScopeContext.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2021 Contributors to Eclipse Foundation.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+package com.sun.faces.cdi.clientwindow;
+
+import jakarta.enterprise.context.ContextNotActiveException;
+import jakarta.enterprise.context.spi.Context;
+import jakarta.enterprise.context.spi.Contextual;
+import jakarta.enterprise.context.spi.CreationalContext;
+import jakarta.enterprise.inject.Typed;
+import jakarta.faces.context.FacesContext;
+import jakarta.faces.lifecycle.ClientWindowScoped;
+
+import java.io.Serializable;
+import java.lang.annotation.Annotation;
+import java.util.logging.Logger;
+
+import static java.util.logging.Level.FINEST;
+import static java.util.logging.Level.SEVERE;
+
+/**
+ * The CDI context for CDI ClientWindowScoped beans.
+ */
+@Typed()
+public class ClientWindowScopeContext implements Context, Serializable
+{
+
+    /**
+     * Stores the logger.
+     */
+    private static final Logger LOGGER = Logger.getLogger(ClientWindowScopeContext.class.getName());
+
+    /**
+     * Stores the serial version UID.
+     */
+    private static final long serialVersionUID = -6559364724281899428L;
+
+    /**
+     * Constructor.
+     */
+    public ClientWindowScopeContext() {
+        LOGGER.log(FINEST, "Creating ClientWindowScope CDI context");
+    }
+
+    /**
+     * Assert the context is active, otherwise throw ContextNotActiveException.
+     */
+    private final void assertNotReleased() {
```

Comment:
unnecessary modifier `final`

Reply:
I´ll do the same cleanup for existing ViewScopeContext.